plutus-conformance/src/PlutusConformance/Common.hs

```diff
@@ -104,21 +40,156 @@ shownParseError = "parse error"
 shownEvaluationFailure :: T.Text
 shownEvaluationFailure = "evaluation failure"
 
--- UPLC evaluation test functions
+-- | The default parser to parse UPLC program inputs.
+parseTxt ::
+    T.Text
+    -> Either ParserErrorBundle (UPLC.Program Name DefaultUni DefaultFun SourcePos)
+parseTxt resTxt = runQuoteT $ UPLC.parseProgram resTxt
 
+-- | The input/output UPLC program type.
 type UplcProg = UPLC.Program Name DefaultUni DefaultFun ()
 
+-- UPLC evaluation test functions
+
+-- | The evaluator to be tested. It should either return a program if the evaluation is successful,
+-- or `Nothing` if not.
 type UplcEvaluator = UplcProg -> Maybe UplcProg
 
--- | A UPLC evaluation test suite.
-data UplcEvaluationTest =
-    MkUplcEvaluationTest {
-       -- | The evaluator function to use for the test.
-       evaluator :: UplcEvaluator
-       -- | The test directory in which the test files are located.
-       , testDir :: FilePath
+-- | The type the tests are comparing.
+data RawTxtOrDebruijnTerm
```

Comment:
The name of this type as it is should be `RawTxtAndDebruijnTerm`, but you only need either the `RawTxt` or the `Debruijn Term`, never both, so you might as well just use `Either T.Text (UPLC.Term NamedDeBruijn ...)`.

Reply:
> The name of this type as it is should be `RawTxtAndDebruijnTerm`, 

True.

> but you only need either the `RawTxt` or the `Debruijn Term`, never both, so you might as well just use `Either T.Text (UPLC.Term NamedDeBruijn ...)`.

That's a bit more awkward to extract the fields than `data` though. But actually I won't need one of the fields now so yeah this part of the code will change anyway.

Reply:
Actually I needed it to be a record because I needed to return both and grab the one I need later. I wonder if it's worth it to make it a record just for readability though.